src/commons/asBaseComponent.tsx

```diff
@@ -25,17 +25,30 @@ function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentT
     static propTypes: any;
     static defaultProps: any;
 
+    state = {
+      error: false
+    };
+
     static getThemeProps = (props: any, context: any) => {
       return Modifiers.getThemeProps.call(WrappedComponent, props, context);
     }
 
+    static getDerivedStateFromError(error: any) {
+      UIComponent.defaultProps.onError(error, WrappedComponent.defaultProps);
```

Comment:
Did you test what happens in public uilib when there's an error? (:

Reply:
Handled now :)